### ISSUE TITLE
MultiRegistration::loadMovables: voxel size is different - MultiRegistration::loadMovables:: Too many arguments

### ISSUE THREAD

**Issue by dmd** [nipreps/fmriprep#2699]
One of my users is getting this on every participant they try:

```
mri_robust_template --satit --fixtp --mov /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise0/clipped.nii.gz /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise1/clipped.nii.gz /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise2/clipped.nii.gz /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise3/clipped.nii.gz --inittp 1 --iscale --noit --template mri_robust_template_out.mgz --subsample 200 --lta /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/epi_merge/tp1.lta /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/epi_merge/tp2.lta /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/epi_merge/tp3.lta /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/epi_merge/tp4.lta
Standard output:
$Id: mri_robust_template.cpp,v 1.54 2016/05/05 21:17:08 mreuter Exp $

--satit: Will estimate SAT iteratively!
--fixtp: Will map everything to init TP!
--mov: Using /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise0/clipped.nii.gz as movable/source volume.
--mov: Using /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise1/clipped.nii.gz as movable/source volume.
--mov: Using /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise2/clipped.nii.gz as movable/source volume.
--mov: Using /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise3/clipped.nii.gz as movable/source volume.
    Total: 4 input volumes
--inittp: Using TP 1 as target for initialization
--iscale: Enableing intensity scaling!
--noit: Will output only first template (no iterations)!
--template: Using mri_robust_template_out.mgz as template output volume.
--subsample: Will subsample if size is larger than 200 on all axes!
--lta: Will output LTA transforms
Setting iscale ...
reading source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise0/clipped.nii.gz'...
converting source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise0/clipped.nii.gz' to bspline ...
MRItoBSpline degree 3
reading source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise1/clipped.nii.gz'...
converting source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise1/clipped.nii.gz' to bspline ...
MRItoBSpline degree 3
reading source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise2/clipped.nii.gz'...
converting source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise2/clipped.nii.gz' to bspline ...
MRItoBSpline degree 3
reading source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise3/clipped.nii.gz'...
converting source '/data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise3/clipped.nii.gz' to bspline ...
MRItoBSpline degree 3
Standard error:
ERROR: MultiRegistration::loadMovables: images have different voxel sizes.
  Currently not supported, maybe first make conform?
  Debug info: size(2) = 3.5, 3.5, 3.5   size(0) = 3.125, 3.125, 5
MultiRegistration::loadMovables: voxel size is different /data/fmriprep-workdir/ndas/fmriprep_wf/single_subject_CB33_wf/syn_preprocessing_auto_00000/epi_reference_wf/clip_bg_noise/mapflow/_clip_bg_noise2/clipped.nii.gz.

Return code: 251
```

The launch is:
```
/usr/bin/singularity run \
    --contain \
    --cleanenv \
    -B /tmp -B /data -B /data1 -B /data2 -B /data3 -B /n -B /cm/shared \
    /cm/shared/singularity/images/fmriprep-21.0.0.simg \
    /data/ndas/00079510 \
    /data/ndas/00079510/derivatives \
    participant \
    --fs-license-file /cm/shared/freesurfer-6.0.1/license.txt \
    --participant_label CB33 \
    --output-spaces MNI152NLin2009cAsym:res-2 anat func fsaverage \
    --n_cpus 4 \
    --mem-mb 8192 \
    --notrack \
    --use-aroma \
    --use-syn-sdc \
    -vv \
    -w  /data/fmriprep-workdir/ndas
```

**Comment by effigies** [comment]
This issue is on https://github.com/nipreps/sdcflows/issues/255. We need to decide what to do here. The simplest solution is to create a separate fieldmap estimate for each voxel size, with the cost of more computation time and (possibly unwanted) variability in correction across runs. Other solutions could be devised to reduce these costs, but may have unwanted tradeoffs.

Care to weigh in over there?

**Comment by effigies** [comment]
This will be resolved by nipreps/sdcflows#312 if merged. Moved to the 22.1.0 milestone until we make a decision.
